onnxruntime/core/providers/cpu/math/element_wise_ops.h

```diff
@@ -6,132 +6,246 @@
 #include "core/common/common.h"
 #include "core/framework/op_kernel.h"
 #include "core/util/math_cpuonly.h"
+#include "core/providers/cpu/element_wise_ranged_transform.h"
 
 namespace onnxruntime {
+namespace functors {
 
-template <typename T>
-class Add final : public OpKernel {
- public:
-  Add(const OpKernelInfo& info) : OpKernel(info) {
+template<typename T>
+struct Log : public ElementWiseRangedTransform<T> {
+  Status Init(const onnxruntime::NodeAttributes) {
+    return Status::OK();
   }
 
-  Status Compute(OpKernelContext* context) const override;
+  ElementWiseRangedTransform<T>* Copy() const {
+    using T1 = typename std::remove_pointer<decltype(this)>::type;
+    using T2 = typename std::remove_const<T1>::type;
+    return new T2(*this);
+  }
+  
+  float Cost() const final { return 15.0f; }
+
```

Comment:
Value needs comment explaining how it was chosen.

Reply:
We started with micro-benchmarks. Then we ran and tuned it on a collection of real benchmarks to obtained the best perf. More details about the benchmarks and results were discussed in an email thread that involved you.

Reply:
Are the micro-benchmarks checked in?  In general it would be good to revisit some of the choices for costs now, particularly when using the Eigen pool after the changes made there since 1.3.  Ideally we could split that work into (1) check that the thread pool is responding correctly to the costs that are provided, and (2) check that reasonable costs are being provided.

Reply:
The micro-benchmarks are just a couple single node models that I created locally.  I didn't check it in since it's a bit trivial. Actually, @snnn plans a much more sophisticated cost tuning process, which I'd help up with.